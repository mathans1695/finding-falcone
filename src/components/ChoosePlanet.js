import React from 'react';
import '../styles/ChoosePlanet.css';

function ChoosePlanet(props) {
	function handleChange(e) {
		const id = e.target.parentElement.id;
		const value = e.target.value;
		let distance;
		
		for (let i=0; i<e.target.options.length; i++) {
			if(e.target.options[i].selected === true) {
				distance = +e.target.options[i].getAttribute('data-distance');
			}
		}
		
		// invokes updatePlanet in MissionPlan component
		props.updateListOfPlanets(id, value, distance);
		props.updateListOfVehicles(id, distance);
	}
	
	const { planets, id } = props.planets;
	const options = planets.map(planet => {
		return <option 
					key={planet.name} 
					value={planet.name}
					className='ChoosePlanet__Option'
					data-distance={planet.distance}
			   >
					{planet.name}
			   </option>
	});
		
	return (
		<div className='ChoosePlanet' id={id}>
			<select
				className='ChoosePlanet__Select'
				value={props.planets.curPlanet}
				onChange={handleChange}
			>
				<option 
					value='Choose Planet' 
					className='ChoosePlanet__Option'
					hidden={true}
				>Choose Planet
				</option>
				{options}
			</select>
		</div>
	)
}

export default ChoosePlanet;