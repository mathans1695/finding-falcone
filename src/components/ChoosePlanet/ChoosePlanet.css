.ChoosePlanet {
	width: 100%;
	padding-left: 5px;
	margin-bottom: 20px;
}

.ChoosePlanet__Select {
	display: block;
	width: 180px;
	padding: 10px 5px;
	border-radius: 5px;
	background-image: linear-gradient(180deg, white, rgba(255,250,240));
	border: none;
	box-shadow: 1px 1px 5px -2px black;
	font-style: sans-serif;
}

.ChoosePlanet__Select:hover {
	cursor: pointer;
	background-image: linear-gradient(180deg, rgba(255,250,240), white);
}