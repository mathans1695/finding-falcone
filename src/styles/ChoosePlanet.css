.ChoosePlanet {
	width: 100%;
	padding-left: 5px;
	margin-bottom: 20px;
}

.ChoosePlanet__Select {
	display: block;
	width: 180px;
	padding: 10px 5px;
	border-radius: 5px;
	background-image: linear-gradient(180deg, white, rgba(0, 0, 0, 0.1));
	font-size: 16px;
	font-weight: 700;
	border: none;
	box-shadow: 1px 1px 5px -2px black;
	font-style: sans-serif;
}

.ChoosePlanet__Select:hover {
	cursor: pointer;
	background-image: linear-gradient(180deg, rgba(0, 0, 0, 0.1), white);
}