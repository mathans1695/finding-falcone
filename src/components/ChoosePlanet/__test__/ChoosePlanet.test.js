import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ChoosePlanet from '../ChoosePlanet';
import { initialPlanets, 
		 changePlanets, 
		 mockUpdateListOfPlanets,
		 mockUpdateListOfVehicles } from './helper';

it('Renders initial rendering correctly', () => {
	const wrapper = shallow(<ChoosePlanet planets={initialPlanets} />);
	
	expect(toJson(wrapper)).toMatchSnapshot();
})

it('Renders correctly onChange', () => {
	const wrapper = shallow(<ChoosePlanet planets={changePlanets} />);
	
	expect(toJson(wrapper)).toMatchSnapshot();
})

it('Select value equals curPlanet property of Planets', () => {
	const wrapper = shallow(<ChoosePlanet planets={changePlanets} />);
	const value = wrapper.find('.ChoosePlanet__Select').props.value;
	const curPlanet = wrapper.props();
	console.log(curPlanet);
	
	expect(value).toEqual(curPlanet);
});