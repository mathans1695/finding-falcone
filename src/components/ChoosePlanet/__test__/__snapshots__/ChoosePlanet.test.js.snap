// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly onChange 1`] = `
<div
  className="ChoosePlanet"
  id="111111"
>
  <select
    className="ChoosePlanet__Select"
    onChange={[Function]}
    value="Donlon"
  >
    <option
      className="ChoosePlanet__Option"
      hidden={true}
      value="Choose Planet"
    >
      Choose Planet
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={100}
      key="Donlon"
      value="Donlon"
    >
      Donlon
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={200}
      key="Enchai"
      value="Enchai"
    >
      Enchai
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={300}
      key="Jebing"
      value="Jebing"
    >
      Jebing
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={400}
      key="Sapir"
      value="Sapir"
    >
      Sapir
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={500}
      key="Lerbin"
      value="Lerbin"
    >
      Lerbin
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={600}
      key="Pingasor"
      value="Pingasor"
    >
      Pingasor
    </option>
  </select>
</div>
`;

exports[`Renders initial rendering correctly 1`] = `
<div
  className="ChoosePlanet"
  id="111111"
>
  <select
    className="ChoosePlanet__Select"
    onChange={[Function]}
    value="Choose Planet"
  >
    <option
      className="ChoosePlanet__Option"
      hidden={true}
      value="Choose Planet"
    >
      Choose Planet
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={100}
      key="Donlon"
      value="Donlon"
    >
      Donlon
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={200}
      key="Enchai"
      value="Enchai"
    >
      Enchai
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={300}
      key="Jebing"
      value="Jebing"
    >
      Jebing
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={400}
      key="Sapir"
      value="Sapir"
    >
      Sapir
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={500}
      key="Lerbin"
      value="Lerbin"
    >
      Lerbin
    </option>
    <option
      className="ChoosePlanet__Option"
      data-distance={600}
      key="Pingasor"
      value="Pingasor"
    >
      Pingasor
    </option>
  </select>
</div>
`;
